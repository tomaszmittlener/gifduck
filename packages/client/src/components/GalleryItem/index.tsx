import React, { FunctionComponent } from 'react'
import styled from 'styled-components'

interface GalleryProps {
  className?: string
  testId?: string
}

const GalleryItem = styled.li`
  display: block;
`

const Gallery: FunctionComponent<GalleryProps> = ({ className, testId, children }) => {
  return (
    <GalleryItem data-testid={testId} className={className}>
      {children}
    </GalleryItem>
  )
}

export default Gallery
