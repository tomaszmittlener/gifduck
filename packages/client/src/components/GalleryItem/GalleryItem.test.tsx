import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Theme from 'styles/theme'

import GalleryItem from './'

const defaultProps = {
  className: 'class-name',
  testId: 'test-id',
}

afterEach(cleanup)

describe('GalleryItem', () => {
  test('renders correctly', async () => {
    const { container } = render(
      <ThemeProvider theme={Theme}>
        <GalleryItem />
      </ThemeProvider>,
    )
    expect(container).toMatchSnapshot()
  })

  test('passes className to container', async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={Theme}>
        <GalleryItem {...defaultProps} />
      </ThemeProvider>,
    )

    expect(getByTestId(defaultProps.testId)).toHaveClass(defaultProps.className)
  })
})
