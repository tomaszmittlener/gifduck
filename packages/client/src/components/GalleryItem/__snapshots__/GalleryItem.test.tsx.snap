// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem renders correctly 1`] = `
.c0 {
  display: block;
}

<div>
  <li
    class="c0"
  />
</div>
`;
